packages/wrangler/src/deployment-bundle/create-worker-upload-form.ts

```diff
@@ -13,31 +13,43 @@ import type {
 } from "./worker.js";
 import type { Json } from "miniflare";
 
+const moduleTypeMimeType: { [type in CfModuleType]: string | undefined } = {
+	esm: "application/javascript+module",
+	commonjs: "application/javascript",
+	"compiled-wasm": "application/wasm",
+	buffer: "application/octet-stream",
+	text: "text/plain",
+	python: "text/x-python",
+	"python-requirement": "text/x-python-requirement",
+	"nodejs-compat-module": undefined,
+};
+
 export function toMimeType(type: CfModuleType): string {
-	switch (type) {
-		case "esm":
-			return "application/javascript+module";
-		case "commonjs":
-			return "application/javascript";
-		case "compiled-wasm":
-			return "application/wasm";
-		case "buffer":
-			return "application/octet-stream";
-		case "text":
-			return "text/plain";
-		case "python":
-			return "text/x-python";
-		case "python-requirement":
-			return "text/x-python-requirement";
-		default:
-			throw new TypeError("Unsupported module: " + type);
+	const mimeType = moduleTypeMimeType[type];
+	if (mimeType === undefined) {
+		throw new TypeError("Unsupported module: " + type);
 	}
+
+	return mimeType;
+}
+
+export function fromMimeType(mimeType: string): CfModuleType {
+	const moduleType = Object.keys(moduleTypeMimeType).find(
+		(type) => moduleTypeMimeType[type as CfModuleType] === mimeType
+	) as CfModuleType | undefined;
+	if (moduleType === undefined) {
+		throw new TypeError("Unsupported module: " + mimeType);
+	}
+
+	return moduleType;
 }
 
 export type WorkerMetadataBinding =
 	// If you add any new binding types here, also add it to safeBindings
 	// under validateUnsafeBinding in config/validation.ts
+	| { type: "inherit"; name: string }
```

Comment:
I decided against it just because it is a weird one to use properly in wrangler, it's much more for actual API usage. 
We have keep_bindings which handles this by type already in wrangler

It just needed to be added here so I didn't have type errors in the API calls

Reply:
can we flag that with a comment then pls 🙏 so as to avoid any future confusions